src/format/parse.rs

```diff
@@ -224,6 +224,103 @@ pub(crate) fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseRes
     Ok((s, ()))
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub(crate) enum Iso8601Format {
```

Comment:
This enum serves no purpose yet. We need it to make sure that the date, time and offset components use the same format when parsing to a `NaiveDateTime`, `DateTime` or duration.